IdentityCore/src/network/request/MSIDAADV1RefreshTokenGrantRequest.m

```diff
@@ -43,8 +52,29 @@ - (instancetype)initWithEndpoint:(NSURL *)endpoint
         NSMutableDictionary *parameters = [_parameters mutableCopy];
         parameters[MSID_OAUTH2_RESOURCE] = resource;
         _parameters = parameters;
+        _thumbprintParameters = [_parameters mutableCopy];
+        _thumbprintParameters[MSID_OAUTH2_REQUEST_ENDPOINT] = endpoint;
+        _thumbprintParameters[MSID_OAUTH2_HOME_ACCOUNT_ID] = ((MSIDRequestParameters *)context).accountIdentifier.homeAccountId;
```

Comment:
don't cast context. It might be not of a MSIDRequestParameters type. 
If something is not available in the request class, it shouldn't be important for thumbprint calculation.

Reply:
Yes indeed, even though that information is encrypted in RT blob.  In that case, it would be best to not include it in the strict request thumbprint calculation.